users/models.py

```diff
@@ -0,0 +1,98 @@
+from django.db import models
+from django.utils import timezone
+from django.core.validators import validate_email, RegexValidator
+from django.contrib.auth.models import make_password
+from django.contrib.auth.models import (
+    AbstractBaseUser, BaseUserManager, PermissionsMixin)
+from django.conf import settings
+
+from common.models import Contact, County, AbstractBase
+
+
+USER_MODEL = settings.AUTH_USER_MODEL
+
+
+class MflUserManager(BaseUserManager):
+    def create_user(self, email, first_name,
+                    username, password=None, **extra_fields):
+        now = timezone.now()
+        if not email:
+            raise ValueError('The email must be set')
+        validate_email(email)
+        p = make_password(password)
+        email = MflUserManager.normalize_email(email)
+        user = self.model(email=email, first_name=first_name, password=p,
+                          username=username,
+                          is_staff=False, is_active=True, is_superuser=False,
+                          last_login=now, date_joined=now, **extra_fields)
+        user.save(using=self._db)
+        return user
+
+    def create_superuser(self, email, first_name, username,
+                         password, **extra_fields):
+        user = self.create_user(email, first_name,
+                                username, password, **extra_fields)
+        user.is_staff = True
+        user.is_active = True
+        user.is_superuser = True
+        user.save(using=self._db)
+        return user
+
+
+class MflUser(AbstractBaseUser, PermissionsMixin):
+    email = models.EmailField(null=False, blank=False, unique=True)
+    first_name = models.CharField(max_length=60, null=False, blank=False)
+    last_name = models.CharField(max_length=60, blank=True)
+    other_names = models.CharField(max_length=80, null=False, blank=True,
+                                   default="")
+    username = models.CharField(
+        max_length=60, null=False,
+        blank=False, unique=True,
+        validators=[RegexValidator(
+            regex=r'^\w+$',
+            message='Preferred name contain only '
+                    'letters numbers or underscores'
+        )
+        ])
+
+    is_staff = models.BooleanField(default=False)
+    is_active = models.BooleanField(default=True)
+    date_joined = models.DateTimeField(default=timezone.now)
+    is_incharge = models.BooleanField(default=False)
+    contact = models.ForeignKey(Contact, null=True, blank=True)
+    county = models.ForeignKey(County, null=True, blank=True)
+    is_national = models.BooleanField(default=False)
```

Comment:
These boolean fields are a noble effort, but I feel that they have not been thought through completely;

Some questions:

 * why is there a `county` FK in the user model, yet there is a join table that links users to counties?
 * what happens if a user leaves a county?
 * what about sub-counties? [ maybe constituencies ]? That will be the first point of data input.
 * why do we have an `is_incharge` field, yet that information is held in the join table that links users to counties?
 * can a user have more than one contact? [ esp. after refactoring the `Contact` model ]
 * how does all this interact with filtering in the serializers / views? Do we need a custom permissions backend? What would be the behavior of that custom permissions backend?

@ngash - Consulting needed here.